### DISCUSSION TITLE
License Change, Commercialisation Plans, and Project Direction Update

### DISCUSSION THREAD

**Discussion by TreyWW** [TreyWW/MyFinances#277] [Announcements]
Hi all,

> Note: A TLDR of this announcement is listed below 

This week we will be changing the license to AGPL v3 due to the reasons stated in #180. This means that users can modify, distribute and use the product commercially still, but cannot do so privately (so must therefore provide source and credit).

Furthermore, I **will** be commercialising this product for the “hosted” version and selling this to some of my existing clients. Now, if anything, this is a very good thing for all of us. It allows me to have funding to actually run this, give users a demo version so you can see your additions actually come to life, and give myself more motivation to work on the project.

Using the AGPL 3.0 License, you too will be able to commercialise the project! The only limitations are that you will have to provide the source and disclose any changes you add to your “forked” copy. In my opinion, this is a win-win for us all. You get to still use the project for both yourself and potential clients, and we get to keep it a community project so all can benefit.

The hosted version will *likely* have two pricing structures: monthly and pay for what you consume. This means that users will have the option to pay a set monthly fee for specific quota usage, or pay for additional usage where needed. This means that users aren't forced to upgrade a whole package just for let's say just more receipt storage. Though, I did say “likely”, this may change as we haven't fully came up with a plan yet alone have any features ready to price anyways.

To be completely honest, we will be adding **closed-source addons** to some new features such as a Billing module and custom features that our clients need. This is mainly for the security side of things - but also because some of the features our clients need isn't totally relevant to the project. This shouldn't be a major deal, but I thought I'd keep things transparent.

Now, as for the direction of the project - it has kind of been all over the place; we have added some non-relevant things to “Finances” and had some really slow development/feature progression. So I apologise for that. This week I will be changing the license, bumping the version, and planning some real features. But to be honest, I **need your help**. I need ideas on how we can make this a worthy product and what features **you** need. At the moment I'm adding features that I would use, but that doesn't fit everyone and is making the whole project go off in a tangent - we need to go back in line.

So what has the project done so far? We've added receipts, invoices, client details management and currency conversion. Not very impressive. I have recently been working on Invoices, adding scheduled invoice sending and reminders for paying invoices, which in my opinion has turned out quite nice. But this isn't really enough. There's no “product” yet. At the moment we've added nothing really to do with “Finances”, no budgeting, no spend tracking etc. The product has kind of aimed to be more of a business management tool, keep track of your customers. Maybe “My**Finances**” is the wrong name?

I'd love to hear feedback of the project, opinions about the changes listed above, and any future ideas you may have of the project and how **you** would like to contribute and what **you** would like to gain out of that.

Thanks for reading, a TLDR is listed below:

### TLDR
- License change to AGPL v3, requiring source disclosure and credit for modifications
- Commercialisation of hosted version to support project development and offer demo capabilities
- Users can also commercialise their versions under AGPL 3.0 with source disclosure
- Hosted version may have monthly and/or pay-per-use pricing structures
- Closed-source addons for security and client-specific features
- Acknowledgment of project direction shift, focus on refining features and your input
- Current features include receipts, invoices, client management, and currency conversion (yikes)
- Plans to enhance finance-related functionalities or change the idea of ”finance”, seeking your input for product direction
- Feedback, suggestions, and contributions from the community are welcome
